src/app/shared/filesystem/file-browser/file-browser.component.html

```diff
@@ -1,28 +1,22 @@
 <div class="top-wrapper">
-  <clark-file-breadcrumb [paths$]="currentPath$"></clark-file-breadcrumb>
+  <clark-file-breadcrumb [paths]="currentPath" (pathChanged)="handlePathChanged($event)"></clark-file-breadcrumb>
   <div class="view-switcher inline-flex right">
     <div class="inline-flex">
-      <button *ngIf="canManage && ((currentNode$| async)?.children.length || (currentNode$| async)?.files.length)" id="new-file" (click)="emitNewOptionClick($event)" class="button new">
+      <button *ngIf="canManage && ((currentNode$| async)?.folders.length || (currentNode$| async)?.files.length)" id="new-file" (click)="emitNewOptionClick($event)" class="button new">
         New Upload <i class="fas fa-upload"></i>
       </button>
-      <!-- <span *ngIf="view==='list'" (click)="view='grid'" class="button clean">
-        <i class="fas fa-th icon-clean"></i>
-      </span>
-      <span *ngIf="view==='grid'" (click)="view='list'" class="button clean">
-        <i class="fas fa-list icon-clean"></i>
-      </span> -->
     </div>
   </div>
 
 </div>
-<div (click)="emitContainerClick($event)" class="flex-wrapper" [ngClass]="{empty: !(currentNode$| async)?.children.length && !(currentNode$| async)?.files.length}">
-  <div *ngIf="canManage && !(currentNode$| async)?.children.length && !(currentNode$| async)?.files.length" class="dropzone-message">
+<div (click)="emitContainerClick($event)" class="flex-wrapper" [ngClass]="{empty: !(currentNode$| async)?.folders.length && !(currentNode$| async)?.files.length}">
+  <div *ngIf="canManage && !(currentNode$| async)?.folders.length && !(currentNode$| async)?.files.length" class="dropzone-message">
     <i class="fal fa-cloud-upload"></i>
     <div class="empty-title">No materials here!</div>
     <button class="button" id="new-file" (click)="emitNewOptionClick($event)">Click here to upload</button>
     <span>or drag and drop files and folders here</span>
   </div>
-  <clark-file-list-view *ngIf="((currentNode$| async)?.children.length || (currentNode$| async)?.files.length)"
+  <clark-file-list-view *ngIf="((currentNode$| async)?.folders.length || (currentNode$| async)?.files.length)"
```

Comment:
Why not use the new isFolderEmpty function here?
```suggestion
  <clark-file-list-view *ngIf="!isFolderEmpty(currentNode$| async)"
```